src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -365,6 +387,93 @@ bool PerformAudioPassThruRequest::IsWaitingHMIResponse() {
   return awaiting_tts_speak_response_ || awaiting_ui_response_;
 }
 
+void PerformAudioPassThruRequest::ProcessAudioPassThruIcon(
+    ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  audio_pass_thru_icon_exists_ = true;
+  if ((*message_)[strings::msg_params].keyExists(
+          strings::audio_pass_thru_icon)) {
+    if (IsAudioPassThruIconParamValid()) {
+      const std::string& icon_storage_file = GetAudioPassThruIconFilename(app);
+
+      if (!file_system::FileExists(icon_storage_file)) {
+        LOG4CXX_WARN(
+            logger_,
+            "Invalid audio_pass_thru_icon doesn't exist in the file system");
+        audio_pass_thru_icon_exists_ = false;
+      }
+    } else {
+      LOG4CXX_WARN(logger_,
+                   "Invalid audio_pass_thru_icon validation check failed");
+      (*message_)[strings::msg_params].erase(strings::audio_pass_thru_icon);
+    }
+  }
+}
+
+mobile_apis::Result::eType
+PerformAudioPassThruRequest::PrepareAudioPassThruResultCodeForResponse(
+    const ResponseInfo& ui_response, const ResponseInfo& tts_response) {
+  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
+
+  hmi_apis::Common_Result::eType common_result =
+      hmi_apis::Common_Result::INVALID_ENUM;
+  const hmi_apis::Common_Result::eType ui_result = ui_response.result_code;
+  const hmi_apis::Common_Result::eType tts_result = tts_response.result_code;
+
+  if ((ui_result == hmi_apis::Common_Result::SUCCESS) &&
+      (tts_result != hmi_apis::Common_Result::SUCCESS)) {
+    common_result = hmi_apis::Common_Result::WARNINGS;
+  } else if (tts_result == hmi_apis::Common_Result::ABORTED) {
+    common_result = hmi_apis::Common_Result::ABORTED;
```

Comment:
@tkireva I think this checking is redundant because above in method PrepareResponseParameters you have checking 
IsAnyHMIComponentAborted